src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -36,41 +38,57 @@
 @WebServlet("/game")
 public final class GameServlet extends HttpServlet {
 
-  private static final String DEVELOPER_KEY = "AIzaSyBZw4Z25Lect7ux9z960RCM7YORcYo6slc";
-
-  private static final String APPLICATION_NAME = "Song Guessing Game";
-  private static final JsonFactory GSON_FACTORY = GsonFactory.getDefaultInstance();
-  private static final int TIME_OFFSET = 3000;
-  private static final int ROUND_LENGTH = 30000;
-  private static final long MAX_RESULTS = 25L;
   private DatastoreService datastore;
+  private Gson gson;
 
   @Override
   public void init() {
     datastore = DatastoreServiceFactory.getDatastoreService();
+    gson = new Gson();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String mostRecentVideoId = getMostRecentVideoId();
-    String json = new Gson().toJson(mostRecentVideoId);
+    Entity game = getCurrentGame();
+    EmbeddedEntity currentRound = getNewRound(game);
+    game.setProperty("currentRound", currentRound);
+    datastore.put(game);
+    EmbeddedEntity currentVideo = (EmbeddedEntity) currentRound.getProperty("video");
+    String currentVideoId = (String) currentVideo.getProperty("videoId");
+    String json = new Gson().toJson(currentVideoId);
     response.getWriter().println(json);
-    // Create a round
-    Entity roundEntity = new Entity("Round");
+  }
+
+  private Entity getCurrentGame() {
```

Comment:
It looks like this method is duplicated in both GameServlet and ChatServlet. As a stretch goal, you could look into moving these duplicate methods into a common class that can be shared across servlets. E.g. an "EntityLoader" class that has methods to load entities from the DataStore.

I think that would be too much for this PR, so could you file an issue for this and assign it to yourself, and link to that issue here? You can pick it up later on if you have time